subsys/net/l2/wifi/wifi_mgmt.c

```diff
@@ -386,6 +425,123 @@ void wifi_mgmt_raise_disconnect_result_event(struct net_if *iface, int status)
 					sizeof(struct wifi_status));
 }
 
+#ifdef CONFIG_WIFI_NM_WPA_SUPPLICANT_ROAMING
+static int wifi_start_roaming(uint32_t mgmt_request, struct net_if *iface,
+			      void *data, size_t len)
+{
+	const struct device *dev = net_if_get_device(iface);
+	const struct wifi_mgmt_ops *const wifi_mgmt_api = get_wifi_api(iface);
+
+	if (roaming_params.is_11r_used) {
+		if (wifi_mgmt_api == NULL ||
+		    wifi_mgmt_api->start_11r_roaming == NULL) {
+			return -ENOTSUP;
+		}
+
+		return wifi_mgmt_api->start_11r_roaming(dev);
+	} else if (roaming_params.is_11k_enabled) {
+		memset(&roaming_params.neighbor_rep, 0x0, sizeof(roaming_params.neighbor_rep));
+		if (wifi_mgmt_api == NULL
+		    || wifi_mgmt_api->send_11k_neighbor_request == NULL) {
+			return -ENOTSUP;
+		}
+
+		return wifi_mgmt_api->send_11k_neighbor_request(dev, NULL);
+	} else if (wifi_mgmt_api == NULL || wifi_mgmt_api->btm_query == NULL) {
+		return -ENOTSUP;
+	}
+
+	return wifi_mgmt_api->btm_query(dev, 0x10);
+}
+
+NET_MGMT_REGISTER_REQUEST_HANDLER(NET_REQUEST_WIFI_START_ROAMING, wifi_start_roaming);
+
+static int wifi_neighbor_rep_complete(uint32_t mgmt_request, struct net_if *iface,
+				      void *data, size_t len)
+{
+	const struct device *dev = net_if_get_device(iface);
+	const struct wifi_mgmt_ops *const wifi_mgmt_api = get_wifi_api(iface);
+	struct wifi_scan_params params = {0};
+	int i = 0;
+
+	for (i = 0; i < roaming_params.neighbor_rep.neighbor_cnt; i++) {
+		params.band_chan[i].channel = roaming_params.neighbor_rep.neighbor_ap[i].channel;
+		if (params.band_chan[i].channel > 14) {
+			params.band_chan[i].band = WIFI_FREQ_BAND_5_GHZ;
+		} else {
+			params.band_chan[i].band = WIFI_FREQ_BAND_2_4_GHZ;
+		}
+	}
+	if (wifi_mgmt_api == NULL || wifi_mgmt_api->candidate_scan == NULL) {
+		return -ENOTSUP;
+	}
+
+	return wifi_mgmt_api->candidate_scan(dev, &params);
+}
+
+NET_MGMT_REGISTER_REQUEST_HANDLER(NET_REQUEST_WIFI_NEIGHBOR_REP_COMPLETE,
+				  wifi_neighbor_rep_complete);
+
+void wifi_mgmt_raise_neighbor_rep_recv_event(struct net_if *iface, char *inbuf, size_t buf_len)
+{
+	const uint8_t *buf = inbuf;
+	char event[32] = {0};
```

Comment:
Please avoid having bare numbers because it is difficult to know what can be written to these. Have a define or something that describes better the value.

If the event is suppose to store ssid, then you need to have one extra byte for `\0`

Reply:
Added new macro define for the max event string length.